gprofiler/utils/node.py

```diff
@@ -0,0 +1,223 @@
+from functools import lru_cache
+import shutil
+import traceback
+from typing import Callable
+from gprofiler.metadata import get_exe_version
+from gprofiler.log import get_logger_adapter
+from granulate_utils.linux.ns import get_proc_root_path, is_same_ns, NsType, run_in_ns
+from granulate_utils.linux.process import is_musl
+
+
+from . import add_permission_dir, pgrep_maps, resource_path, pgrep_exe
+
+import json
+import requests
+import os
+import signal
+import psutil
+import stat
+
+from retry import retry
+from websocket import create_connection
+from websocket._core import WebSocket
+
+logger = get_logger_adapter(__name__)
+
+
+class NodeDebuggerUrlNotFound(Exception):
+    pass
+
+
+class NodeDebuggerUnexpectedResponse(Exception):
+    pass
+
+
+def _start_debugger(pid):
+    os.kill(pid, signal.SIGUSR1)
+
+
+def get_node_processes():
+    return pgrep_maps(r"(?:^.+/node[^/]*$)")
+
+
+@retry(NodeDebuggerUrlNotFound, 5, 1)
+def _get_debugger_url() -> str:
+    # when killing process with SIGUSR1 it will open new debugger session on port 9229,
+    # so it will always the same
+    port = 9229
+    debugger_url_response = requests.get(f"http://127.0.0.1:{port}/json/list")
+    if (
+        debugger_url_response.status_code != 200
+        or not "application/json" in debugger_url_response.headers.get("Content-Type")
```

Comment:
Copying from https://github.com/Granulate/gprofiler/pull/438#discussion_r944920950

> If Content-Type doesn't exist then you have "application/json" in None which will raise

Reply:
Well, this will raise but the exception is caught in `generate_map_for_node_processes` so maybe that's okay..